add files and documentation

rcooke-warwick opened the pull request:
Change-type: minor
Signed-off-by: Ryan Cooke <ryan@balena.io>

rcooke-warwick committed:
Add initial files and documentation

fix diagram and update design based on prototype

Update commands to consider serial number

Point link to new CAD design

Update testing instructions

Update spec.json to new CAD link

Add assembly instructions

Misc proofreading

Update BoM

Fix README.md block diagram

Change-type: minor
Signed-off-by: Pedro Henrique Kopper <pedro.kopper@balena.io>

anujdeshpande commented:
hey @rcooke-warwick - for now - while the lab is being set up - people will be expected to manufacture a lot of their parts on their own locally. Can you just ensure that manufacturing_lab@balena.io is in the loop with any communication? This will help the lab keep track of what was built, and more importantly how.

cc @konmouz

anujdeshpande commented:
@rcooke-warwick have you placed an order for the components from the bom.md file already?

konmouz commented:
@anujdeshpande not sure what you mean by 'is in the loop with any communication'

anujdeshpande commented:
@konmouz whatever improvements we add to the manufacturing lab inbox, will be available to everyone else. For example, if we enable JF threads for that inbox, folks can use that - even if its not a manufacturing lab order.
They can check and set status, assign a person, and so on.

konmouz commented:
@anujdeshpande agreed

phckopper commented on source/3d/onshape.md:
```diff
@@ -0,0 +1,11 @@
+
+The design was created using [onshape](https://www.onshape.com/en/) which is a cloud based cad platform.
+
+The current version of the design can be accessed [here](https://cad.onshape.com/documents/4a734179a4bf72ca3cf4dc82/v/0fcfc861c56588e719f614ad/e/0323c8faa4d42ca9fcdb2b1a?renderMode=0&uiState=628bbdd0a1042f44aeda13e6)
```
https://cad.onshape.com/documents/34e52f69664fd15550ba35ad/w/5abb2d265c5abd27c227e7a7/e/b3a911c86048cfb278f7aa6c?renderMode=0&uiState=62cf7d917c0f5407a2eca93e 

New proposed design. 
Changes:
- Thickness = 2.4mm (easier to print with a 0.8mm nozzle)
- Added M3 screw holes for the C13/C14 connectors
- Added snap-on tab underneath the USB connector for extra support
- Added chamfer to the existing standoffs to improve adhesion with the bottom plate

konmouz replied:
@phckopper I have some comments on the case:

_

<img width="1262" alt="Screenshot 2022-07-14 at 17 29 54" src="https://user-images.githubusercontent.com/15323961/179007453-03195c39-19f1-450a-b627-e189263278fb.png">

There isn't much benefit in having the opening in both top and bottom part. If you push your lines you could keep all openings in the on part and have the other just restricting the open end. Also, by doing this you could use snap on C13/C14 modules while having the case open. Btw, why are we using screw on panel mount?

_

<img width="663" alt="Screenshot 2022-07-14 at 17 30 32" src="https://user-images.githubusercontent.com/15323961/179007494-919e9852-bb36-4f4b-b216-546c15f11f44.png">

Better remove this overhand to avoid drama in printing, There's no harm to have this section going all the way down so it's self supported. Also, no need to use the two small pins since the port opening restricts the USB module in 2 dimensions. 

_

<img width="1385" alt="Screenshot 2022-07-14 at 17 31 09" src="https://user-images.githubusercontent.com/15323961/179007522-73057827-fbdc-491c-814a-d09ff38fe651.png">

Here you can avoid using screws. Instead of 2 pillars, you could have a rect pillar on each end of the case that restricts the USB module from moving up/down or going back/ forth

_

<img width="1440" alt="Screenshot 2022-07-14 at 17 29 26" src="https://user-images.githubusercontent.com/15323961/179007405-572a12dd-2e26-4bb5-a4a0-8dac1f2f790d.png">

Better use max 2mm radius on the fillet so it prints nicely without support. The adjacent edge can remain 5mm or it could even be larger.

_

Finally, I'd use the shorter part as bottom, since it also has the screws exposed. Doesn't change things probably, just bare it on mind. Happy to have a call if all these don't make sense.

phckopper replied:
> There isn't much benefit in having the opening in both top and bottom part. If you push your lines you could keep all openings in the on part and have the other just restricting the open end. Also, by doing this you could use snap on C13/C14 modules while having the case open.

Okay, will update to split the case at the top instead of the middle

> Btw, why are we using screw on panel mount?

I added the screw mounts as C13/C14 are much easier to source here in Brazil and they're compatible with the snap on ones. I figured that it'd be nice to leave the case open to alternatives.

> Better remove this overhand to avoid drama in printing, There's no harm to have this section going all the way down so it's self supported. Also, no need to use the two small pins since the port opening restricts the USB module in 2 dimensions.

Okay, will simplify this part.

> Here you can avoid using screws. Instead of 2 pillars, you could have a rect pillar on each end of the case that restricts the USB module from moving up/down or going back/ forth

Nice idea, I'll try to implement it and add it to the next revision.

> Better use max 2mm radius on the fillet so it prints nicely without support. The adjacent edge can remain 5mm or it could even be larger.

Noted.

phckopper replied:
From autokit check-in:

- Make top lid thinner
- Add plastic piece that holds the relay vertically in place to the top lid

phckopper replied:
![photo_5093862957615524359_y](https://user-images.githubusercontent.com/4000637/179638601-76324953-edc4-4489-b8f2-8df5d495f8a0.jpg)

I've prototyped the bottom part already, the USB Relay fits very snugly. The rectangular pillars were a little bit too short, so I've added an extra 0.5mm to the model. Besides that, it all went great.

phckopper replied:
![IMG_20220721_204942](https://user-images.githubusercontent.com/4000637/180496783-6fe168a7-8c2c-4311-9899-d7d126daa9bb.jpg)
![IMG_20220721_210055](https://user-images.githubusercontent.com/4000637/180496793-4d23ce6a-9b84-4559-9236-2a6b0baf8b25.jpg)
![IMG_20220721_210102](https://user-images.githubusercontent.com/4000637/180496797-6e176e02-3603-4e83-acf5-46696bcaced1.jpg)

phckopper replied:
Final assembled and tested unit, worked out great.

Notes:

- Maybe it'd be interesting to consolidate all screws to M2.5, instead of using both M3 and M2.5. I'll consider this a stretch goal.
- I've made the top lid much thinner, from ~5mm to 2.4mm. This decreased print time dramatically while retaining strength.
- Wiring was tight with 1.5mm^2 wires, but doable. The enclosure should not be reduced further without considering the bending radius.
- Brown wire was near impossible to find, so I used red in my assembly. Further builds should use brown to respect IEC code.
- My build used screw-on connectors instead of the snap-fit ones. Instead of using 6.3mm connectors, they use 4.8mm, so I temporarily soldered them while the correct connectors don't arrive.

rcooke-warwick commented:
great work @phckopper ! I like the changes that you have made, and consolidating the screws to all be the same would be a big improvement imo. 

@Bucknalla @konmouz  - small thing here, but as this design is on onshape, we've put a link to the relevant onshape commit in this repo. Its currently in 2 places, the `spec.json` and here: https://github.com/balena-io-hardware/autokit-relay/blob/ryan/add-initial-files/source/3d/onshape.md

Any ideas on which one is preferred in this case?

rcooke-warwick commented:
Another thing we can look at in the future is reverse engineering the `usb-relay` tool that we're using to toggle the relay and implement it ourselves in typescript using: https://www.npmjs.com/package/node-hid. We use typescript for a lot of things at balena, and most of the autokit software interface will be/is being implemented in it too. 

Having our own npm package to control this thing might be nicer in the long run than relying on us installing a linux package. This is a stretch goal however and shouldn't prevent us from releasing the first version of the product :)

rcooke-warwick commented:
I think another thing that will be very useful going forward is to add mounting holes that would let the relay be mounted onto a panel. Small extrusions from the bottom part with holes would be enough I think, sort of looking like this:
![Screenshot from 2022-07-25 11-35-54](https://user-images.githubusercontent.com/24698912/180758118-bfa68750-afd6-4573-81cb-6e9a15dcf68b.png)

phckopper commented:
@rcooke-warwick maybe we can fork [this](https://github.com/josephdadams/USBRelay) and add type annotations?

phckopper commented:
Created fork here https://github.com/balena-io-modules/node-usbrelay

rcooke-warwick commented:
Nice that's a great find. You can add a `repo.yml` file to that repo like this: https://github.com/balena-io-hardware/testbotsdk/blob/master/repo.yml, and iirc the balena CI will publish an npm pacakge automatically

rcooke-warwick commented:
heey @phckopper could we update the BoM with approximate cable lengths requierd for each of the 4 cables that are used inside the enclosure - and also add the colours for each one. This is so we can give the BoM to Andreas in Athens to make us 10 of these devices

phckopper commented:
@rcooke-warwick these informations can be found in [this commit](https://github.com/balena-io-hardware/autokit-relay-3d/commit/2e202b47d2a362aa9be64522ba0c21fef101ede0)

zisis00 commented:
@rcooke-warwick, @phckopper 
For production will be good to use a DPDT relay that can cut off both AC phases (L&N). In this case use terminals with pitch 7.62mm to be compatible with safety standards.

Also, you can use power entry modules for pcb to avoid the cables and the friction during assembly and sourcing, for example:https://www.amazon.com/Power-Entry-Modules-Mount-piece/dp/B00M1K8216. In this case the pcb should be larger and re-design the case to match the height. I think it worth as the total cost and the complexity are less.

rcooke-warwick commented:
Nice one on opening the issues @phckopper  - could you squash the commits in this PR to a single commit: https://www.git-tower.com/learn/git/faq/git-squash then we can merge it.

phckopper commented:
@rcooke-warwick done

rcooke-warwick commented on source/specification/spec.json:
```diff
@@ -1,7 +1,26 @@
+    "description": "3d printed enclosure for USB relay",
     "hwType" : "3d",
+    "links":{
+        "source": "https://cad.onshape.com/documents/34e52f69664fd15550ba35ad/w/5abb2d265c5abd27c227e7a7/e/b3a911c86048cfb278f7aa6c?renderMode=0&uiState=62dad8318237a701f534681a"
```
@phckopper is this the latest version of the design in onshape? If so I'll use this to mae the 10x units

phckopper replied:
@rcooke-warwick yes

rcooke-warwick commented:
Also, the STL/STEP files in the outputs folder - maybe you should delete them, I dont think we should keep those "output" files in the repo like this anyway any more

rcooke-warwick commented:
@balena-ci retest

Commit:
Remove extra output files

Change-type: patch
Signed-off-by: Pedro Henrique Kopper <pedro.kopper@balena.io>

rcooke-warwick commented:
hey @anujdeshpande we can't seem to merge this as the hardware test check is still pending - I replied to the email but it looks as though something has goeetn lost somewhere

konmouz committed:
case design update

- Reduced the overall size of the case
- Reduced the size of the C13/C14 holes - the screw can self thread
- Made an opening under the PCB for the solder pins to not collide with the case
- The bottom part is designed to push the relay against the top part and lock it in place with no screws
- Added 4 pegs at the bottom for alignment with the top
- Changed the top and bottom corner radiuses so it prints without support reliably
Change-type: minor

konmouz commented:
@rcooke-warwick @phckopper, I pushed a new commit with the case changes. Can you please print and test it to check if I missed anything?

phckopper commented:
@konmouz @rcooke-warwick printed and tested with Skadis mounting holes. Will update the repo with new BOM. 

Remarks:
- C13/C14 screw holes were a little bit too small, even for self-tapping screws, but they still worked with some extra force
- With the extra length of the crimp connectors, the wiring was very tight and we're probably at the limit of the 1.5mm^2 stranded wire bending radius. I reduced the lengths of the cables accordingly, but it's still tight.
- The USB relay was snug and tight without using screws.

![IMG_20220823_190827](https://user-images.githubusercontent.com/4000637/186285959-cb05a60d-6221-4708-8a10-697d6fd16a62.jpg)

![IMG_20220823_191138](https://user-images.githubusercontent.com/4000637/186285971-13b7df25-cb68-4e78-a34c-a499f69bef52.jpg)

Commit:
case design update

- Increased width for better strain relief
- Added Skadis mounting tabs
- Increased screw-on connector hole size to 2mm

Change-type: minor

Signed-off-by: Pedro Henrique Kopper <pedro.kopper@balena.io>

phckopper commented:
@anujdeshpande is there any way to merge this PR forcefully? It's getting too big and I wanted to create another building upon this one.

rcooke-warwick commented:
@phckopper looks like i can merge it
